lua/autorun/nl_library.lua

```diff
@@ -0,0 +1,476 @@
+	net.WriteEntity(ply)
+	net.WriteString(stuntype)
+	net.Broadcast()
+	
+	return true
+end
+
+function NLL.UnRagdoll( ply )
+	if not SERVER then return end
+	local ragvalid = IsValid(ply.nllragdoll)
+	local pos
+	if ragvalid then // Sometimes the ragdoll is missing when we want to unrag, not good!
+		if ply.nllragdoll.hasremoved then return end // It has already been removed.
+		
+		pos = ply.nllragdoll:GetPos()
+		ply:SetModel(ply.nllragdoll:GetModel())
+		ply.nllragdoll.hasremoved = true
+	else
+		pos = ply.nlldroppos // Put him at the place he got sleepd, works great.
+	end
+	ply:SetParent()
+	
+	NLL.PlayerSetPosNoBlock(ply, pos, {ply, ply.nllragdoll})
+	
+	ply:SetCollisionGroup(COLLISION_GROUP_PLAYER)
+	
+	timer.Simple(0,function()
+		SafeRemoveEntity(ply.nllragdoll)
+		NLL.PlayerInvis(ply, false)
+	end)
+	
+	timer.Simple(.1, function()
+		NLL.PlyRetrieveWeapons(ply)
+	end)
+	
+	net.Start("nllendview")
+	net.Send(ply)
+end
+
+
+hook.Add("EntityTakeDamage", "NLLDamageHandler", function(ent, dmginfo)
+	if ent:IsPlayer() and IsValid(ent.nllragdoll) and not ent.ragdolldamage then // If we're hitting the player somehow we won't let, the ragdoll should take the damage.
+		dmginfo:SetDamage(0)
+		return
+	end
+	
+	if IsValid(ent.nllplayer) and (dmginfo:GetAttacker() != game.GetWorld()) then //Worldspawn appears to be very eager to damage ragdolls. Don't!		
+		local ply = ent.nllplayer
+		//To prevent infiniteloop and other trickery, we need to know if it was ragdamage.
+		ply.ragdolldamage = true
+		ply:TakeDamageInfo(dmginfo) // Apply all ragdoll damage directly to the player.
+		ply.ragdolldamage = false
+	end
+end)
+
+hook.Add( "PlayerUse", "NLLRagdollWakeAttempt", function( ply, ent )
+    if IsValid(ent) and IsValid(ent.nllplayer) then //Don't wake players that don't have their condition set to "ZZZ"
+        if ent.nllplayer.condition != "ZZZ" and not ply:IsWalking() then return end // allow 'waking' if the victim is stunned and needs to be killed
+        plytarg = ent.nllplayer
+        NLL.UnRagdoll(plytarg)
+	plytarg.nllismuted = false
+        if timer.Exists("NLLUnragdoll".. plytarg:UserID()) then
+            timer.Remove( "NLLUnragdoll".. plytarg:UserID())
+        end
+        if not SERVER then return end
+        
+	if ply:IsWalking() then // alt + e to kill instead of waking
+	    if ply.CovertBuddha == nil then ply.CovertBuddha = 0 end
+	    if ply.CovertBuddha == 1 then
+		plytarg:SetHealth(1)
+		return
+	    end
+            local dmgtable = DamageInfo()
+            dmgtable:SetDamage((plytarg:Health())*1.5)
+            dmgtable:SetAttacker(ply)
+            dmgtable:SetDamageForce(Vector(0,0,-1))
+            dmgtable:SetDamagePosition(plytarg:GetPos())
+            dmgtable:SetDamageType(4)
+            dmgtable:SetInflictor(ply)
+            timer.Create("NLLKillTarget".. plytarg:UserID(), 0.2, 1, function()
+                plytarg:TakeDamageInfo( dmgtable )
+            end)
+        end
+    end
+end)
```

Comment:
These two hooks are serverside only, you may as well only define them on the server